<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>927</width>
    <height>1068</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Game of Life</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Cell Size</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="horizontalSlider">
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>7</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Grid Width</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBox_w">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Grid Height</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="spinBox_h">
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Frame Time</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSlider" name="horizontalSlider_speed">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="cacheMode">
         <set>QGraphicsView::CacheNone</set>
        </property>
        <property name="optimizationFlags">
         <set>QGraphicsView::DontSavePainterState</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>927</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuGame">
    <property name="title">
     <string>Game</string>
    </property>
    <addaction name="actionReset"/>
   </widget>
   <addaction name="menuGame"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>cellSizeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_w</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>gridWidthChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>281</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_h</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>gridHeightChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_speed</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>simulationSpeedChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>521</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>cellSizeChanged(int)</slot>
  <slot>gridWidthChanged(int)</slot>
  <slot>gridHeightChanged(int)</slot>
  <slot>simulationSpeedChanged(int)</slot>
  <slot>reset()</slot>
 </slots>
</ui>
